File: Project_Sourcecode_go/state/Overdrawn.go
package state

type Overdrawn struct {
	State
}

func (s *Overdrawn) Logout() {
	s.M.ChangeState(1)
}

func (s *Overdrawn) Deposit() {
	s.M.P.MakeDeposit()
	s.M.ChangeState(4)
}
func (s *Overdrawn) IncorrectLock() {
	s.M.P.IncorrectLockMsg()
}
func (s *Overdrawn) Lock() {
	s.M.ChangeState(6)
}
func (s *Overdrawn) Withdraw() {
	s.M.P.MakeWithdraw()
}
func (s *Overdrawn) Balance() {
	s.M.P.DisplayBalance()
}

func (s *Overdrawn) GetName() string {
	return "Overdrawn"
}

File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/state/Overdrawn.java
package edu.iit.cs586.project.state;

import edu.iit.cs586.project.MDA_EFSM;
import edu.iit.cs586.project.output.OP;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/1 22:30
 */
public class Overdrawn extends State{
    public Overdrawn(MDA_EFSM m) {
        super(m);
    }

    @Override
    public String getName() {
        return "Overdrawn";
    }

    @Override
    public void balance() {
        m.p.displayBalance();
    }

    @Override
    public void incorrectLock() {
        m.p.IncorrectLockMsg();
    }

    @Override
    public void lock() {
        this.m.changeState(6);
    }

    @Override
    public void withdraw() {
        m.p.makeWithdraw();
    }

    @Override
    public void logout() {
        this.m.changeState(1);
    }

    @Override
    public void deposit() {
        m.p.makeDeposit();
        m.changeState(4);
    }
}
